ecs-cli/modules/cli/local/secrets/clients/clients.go

```diff
@@ -17,22 +17,32 @@ package clients
 import (
 	"strings"
 
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/config"
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/arn"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
 	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/aws/aws-sdk-go/service/ssm/ssmiface"
 	"github.com/pkg/errors"
 )
 
+// region represents an AWS region.
+type region string
+
 // SSMDecrypter represents a SSM client that implements the secrets.SecretDecrypter interface.
 type SSMDecrypter struct {
-	*ssm.SSM
+	ssmiface.SSMAPI
```

Comment:
I am wondering why you use SSMAPI here instead of SSM. Is there any good to use SSMAPI?

Reply:
Good question! Take a look at this article on how to mock the AWS Go SDK: https://aws.amazon.com/blogs/developer/mocking-out-then-aws-sdk-for-go-for-unit-testing/
If it doesn't make sense, let me know :)

The rough explanation is that `SSMDecrypter` implements all the methods defined by the `ssmiface.SSMAPI` interface. Since we changed the embedding from a concrete type `ssm.SSM` to an interface `ssmiface.SSMAPI` it allows us to swap the underlying object from a real client to a mock client.